moveit_core/constraint_samplers/src/default_constraint_samplers.cpp

```diff
@@ -565,6 +555,26 @@ bool constraint_samplers::IKConstraintSampler::sampleHelper(robot_state::RobotSt
       return false;
     }
 
+    // we now have the transform we wish to perform IK for, in the planning frame
+    if (transform_ik_)
+    {
+      // we need to convert this transform to the frame expected by the IK solver
+      // both the planning frame and the frame for the IK are assumed to be robot links
```

Comment:
Poses are sampled in MoveIt's planning frame.
Depending on your setup&platform the base frame for the IK plugin might be your robot base link, but might also be the base link of the manipulator that is handled by the plugin.
If you have an arm on top of a mobile base, these can be different.